### PR TITLE
chore: drop Node 10 support

### LINKED ISSUE #621: Drop Node 10 support
Because it's [EOL](https://nodejs.org/en/about/releases).

### PR DESCRIPTION
**Pull request by mpeyper**
**What**:

Drop node 10 support

**Why**:

Fixes #621 

**How**:

Remove node `10.13` and `10` from node version matrix in github actions.

**Checklist**:

- [x] Ready to be merged

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     if: ${{ !contains(github.head_ref, 'all-contributors') }}
     strategy:
       matrix:
-        node: [10.13, 10, 12, 14, 16]
+        node: [12, 14, 16]
```
Comment (mpeyper) [inline .github/workflows/validate.yml:19]: Is there be a minimum version of v12 we should be testing against?

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -48,8 +48,8 @@ jobs:
     runs-on: ubuntu-latest
     if:
       ${{ github.repository == 'testing-library/react-hooks-testing-library' &&
-      contains('refs/heads/main,refs/heads/beta,refs/heads/next,refs/heads/alpha',
-      github.ref) && github.event_name == 'push' }}
+      contains('refs/heads/main,refs/heads/beta,refs/heads/next,refs/heads/alpha', github.ref) &&
+      github.event_name == 'push' }}
```
Comment (mpeyper) [inline .github/workflows/validate.yml:52]: I think this was an auto format.

File: .github/workflows/validate.yml (hunk 1 above)
Comment (MichaelDeBoey) [reply .github/workflows/validate.yml:19]: `12.13` was first LTS version

https://nodejs.org/en/blog/release/v12.13.0

```suggestion
        node: [12.13, 12, 14, 16]
```

Comment (MichaelDeBoey) [reply .github/workflows/validate.yml:19]: But we can also check against Node 12.0 if we want

```suggestion
        node: ['12.0', 12, 14, 16]
```

**Review by MichaelDeBoey — APPROVED** [review]

**Review by joshuaellis — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -4,6 +4,9 @@
   "description": "Simple and complete React hooks testing utilities that encourage good testing practices.",
   "main": "lib/index.js",
   "types": "lib/index.d.ts",
+  "engines": {
+    "node": ">=12"
+  },
```
Comment (mpeyper) [inline package.json:9]: I figured if we were going to make a breaking change, we should at least have something breaking in there 😅

**Comment by mpeyper** [comment]
Can somebody just rereview this and check the [commit message](https://github.com/testing-library/react-hooks-testing-library/pull/623/commits/da3dedb814643ac5c4254a9acfd07fd8b043e488) is correct for a major bump.  I always seem to mess up the order of the breaking changes, fixes and co-authors an either the wrong version gets published or the release notes get all messed up.

**Comment by joshuaellis** [comment]
It _should_ cause a breaking change release because it's in the body. I couldn't say about release notes, they're still a mystery to me.

**Comment by mpeyper** [comment]
I found [this](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-footer) which was linked from the semantic-release docs which suggests I need 2 blank lines after the breaking change summary otherwise it will think it is the breaking change description...  I'll put an extra one in to be safe 🤷‍♂️

**Review by joshuaellis — APPROVED** [review]
Yeah the breaking change system always gets me :(

**Comment by mpeyper** [comment]
Annoyingly, git is trimming the double blank line into a single new line when I commit... oh well, guess I'll just have to double check the release notes after it runs.

**Comment by joshuaellis** [comment]
I didn't double line break yesterday and it worked for me if incl release notes...

**Pull request merged**